vsintegration/src/FSharp.Editor/LanguageService/FSharpProjectOptionsManager.fs

```diff
@@ -203,7 +203,11 @@ type private FSharpProjectOptionsReactor (workspace: Workspace, settings: Editor
                             | Some(_, projectOptions) -> referencedProjects.Add(FSharpReferencedProject.CreateFSharp(referencedProject.OutputFilePath, projectOptions))
                         else
                             let! comp = referencedProject.GetCompilationAsync(ct) |> Async.AwaitTask
-                            referencedProjects.Add(createPEReference referencedProject comp ct)
+                            try
+                                let peRef = createPEReference referencedProject comp ct
+                                referencedProjects.Add(peRef)
+                            with
+                            | _ -> ()
```

Comment:
Hmm, I don't know about this. We should allow any exception like that to bubble up somehow IMO. Otherwise we might further propagate issues like this: https://github.com/dotnet/fsharp/issues/9575

The issue there is that we're crashing somehow, but an exception was swallowed, so there's no traceability.

Reply:
I would like to allow the exception to get propagated, but we can't have it take down VS if it does.

Reply:
Right, but that might happen anyways. In the issue I linked, an exception is _not_ propagated, but because of some unknown reason, it causes a corruption in Roslyn that ultimately crashes VS anyways. Except this time there's no stack trace with any causality.

Reply:
I think we can get rid of the try/with safely. When emitting the compilation, I'm setting `tolerateErrors` to `true` which means:
```csharp
/// <summary>
/// Tolerate errors, producing a PE stream and a success result even in the presence of (some) errors. 
/// </summary>
```
It sounds like it doesn't guarantee a success result and a PE stream, but it will do it as best as it can.